services/expiration/expiration-notification.server.ts

```diff
@@ -186,10 +189,12 @@ const getExpiringDnsRecords = () => {
   });
 };
 // function to add jobs
-async function addExpirationNotifications() {
-  let jobName = `${expirationNotificationQueueName}`;
-  return expirationNotificationQueue.add(jobName, {
+async function addExpirationNotifications(type: string) {
+  // we never want to add multiple jobs for the same user/email notification (unique jobId)
+  let jobId = `${expirationNotificationQueueName}-${type}-expiry`;
+  return expirationNotificationQueue.add(expirationNotificationQueueName, {
     repeat: { every: 5 * 60 * 1000 },
+    jobId,
```

Comment:
I don't think this is right.  The [signature for .add()](https://api.docs.bullmq.io/classes/Queue.html#add) takes a `name`, `data`, and `options`.  Here you are putting the `jobId` into the `options`, but I think it needs to go in `data`:

```ts
return expirationNotificationQueue.add(expirationNotificationQueueName, { jobId }, {
  repeat: { every: 5 * 60 * 1000 },
});
```

Test this to make sure I'm right, but this is what I see in the docs.